File: cse-analyzer/frontend/src/api/predictions.ts
import client from "./client";
import type { PredictionRequest, PredictionStatus, PredictionRecord } from "../types";

export const generatePrediction = (body: PredictionRequest) =>
  client.post<{ task_id: string; status: string }>("/predictions/generate", body).then((r) => r.data);

export const getPredictionStatus = (taskId: string) =>
  client.get<PredictionStatus>(`/predictions/status/${taskId}`).then((r) => r.data);

export const getPredictionHistory = () =>
  client.get<PredictionRecord[]>("/predictions/history").then((r) => r.data);

File: cse-analyzer/backend/app/routers/predictions.py
from fastapi import APIRouter, Depends
from app.models.prediction import PredictionRequest, PredictionStatus
from app.services import prediction_service
from app.middleware.auth_middleware import get_current_user

router = APIRouter(prefix="/predictions", tags=["predictions"])


@router.post("/generate")
async def generate_prediction(
    body: PredictionRequest,
    current_user: dict = Depends(get_current_user),
):
    task_id = await prediction_service.enqueue_prediction(
        body.symbol, body.model_type, body.forecast_days, current_user["user_id"]
    )
    return {"task_id": task_id, "status": "pending"}


@router.get("/status/{task_id}", response_model=PredictionStatus)
async def prediction_status(
    task_id: str,
    current_user: dict = Depends(get_current_user),
):
    return await prediction_service.get_prediction_status(task_id)


@router.get("/history")
async def prediction_history(
    limit: int = 20,
    current_user: dict = Depends(get_current_user),
):
    return await prediction_service.get_prediction_history(current_user["user_id"], limit)
